From 05bd0d86ebfd25ff0d4dfa97323739628aa6aa63 Mon Sep 17 00:00:00 2001
From: Yuvasree Sivasankaran <quic_ysivasan@quicinc.com>
Date: Mon, 11 Dec 2023 16:11:56 +0530
Subject: [PATCH] wifi: ath12k: Advertise TX_QUEUE mac hw flag

To avoid tx queuing in mac80211, advertise TX_QUEUE mac hw flag
which enable tx queuing in driver and avoid performance degradation.

Signed-off-by: Yuvasree Sivasankaran <quic_ysivasan@quicinc.com>
---
 drivers/net/wireless/ath/ath12k/mac.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/net/wireless/ath/ath12k/mac.c b/drivers/net/wireless/ath/ath12k/mac.c
index 2904926..8349dfb 100644
--- a/drivers/net/wireless/ath/ath12k/mac.c
+++ b/drivers/net/wireless/ath/ath12k/mac.c
@@ -16654,6 +16654,7 @@ static int ath12k_mac_hw_register(struct ath12k_hw *ah)
 	ieee80211_hw_set(hw, USES_RSS);
 	ieee80211_hw_set(hw, SUPPORTS_TID_CLASS_OFFLOAD);
 	ieee80211_hw_set(hw, SUPPORTS_MULTI_BSSID);
+	ieee80211_hw_set(hw, HAS_TX_QUEUE);
 
 	if (ath12k_frame_mode == ATH12K_HW_TXRX_ETHERNET) {
 		ieee80211_hw_set(hw, SUPPORTS_TX_ENCAP_OFFLOAD);
-- 
2.34.1

